From 03d197cd74d8d48f3ccd2a00dba68bb23d69d0c5 Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@freescale.com>
Date: Mon, 28 Nov 2016 13:37:36 +0800
Subject: [PATCH 1318/5242] MLK-13574-1: ASoC: imx-wm8960: remove the gpr
 dependency

commit  6f52b7e85ea7ea26f51d2dbae7a58e698bfa0ddc from
https://source.codeaurora.org/external/imx/linux-imx.git

There is no gpr setting in some board. so we can't return
a fatal error.

Signed-off-by: Shengjiu Wang <shengjiu.wang@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/imx-wm8960.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/sound/soc/fsl/imx-wm8960.c b/sound/soc/fsl/imx-wm8960.c
index 4a8150a..7ed03fb 100644
--- a/sound/soc/fsl/imx-wm8960.c
+++ b/sound/soc/fsl/imx-wm8960.c
@@ -481,8 +481,7 @@ static int imx_wm8960_probe(struct platform_device *pdev)
 	ret = of_parse_phandle_with_fixed_args(pdev->dev.of_node, "gpr", 3,
 				0, &args);
 	if (ret) {
-		dev_err(&pdev->dev, "failed to get gpr property\n");
-		goto fail;
+		dev_warn(&pdev->dev, "failed to get gpr property\n");
 	} else {
 		data->gpr = syscon_node_to_regmap(args.np);
 		if (IS_ERR(data->gpr)) {
-- 
1.7.9.5

